app/services/import_polygons.rb

```diff
@@ -1,67 +1,80 @@
 class ImportPolygons
   BUFFER_API_URL = "https://tasks.arcgisonline.com/arcgis/rest/services/Geometry/GeometryServer/buffer?".freeze
+  BUFFER_DISTANCES_IN_MILES = [5, 10, 15, 20, 25].freeze
   URL_MAXIMUM_LENGTH = 33_000
 
   include DistanceHelper
 
-  attr_reader :location_type
+  attr_reader :location_name, :api_location_type
 
-  def initialize(location_type:)
-    @location_type = location_type
+  def initialize(api_location_type:)
+    @api_location_type = api_location_type
   end
 
   def call
-    response = JSON.parse(HTTParty.get(LOCATION_POLYGON_SETTINGS[location_type][:boundary_api]))
+    response = JSON.parse(HTTParty.get(LOCATION_POLYGON_SETTINGS[api_location_type][:boundary_api]))
 
     response.fetch("features", []).each do |region_response|
-      location_name = region_response.dig("attributes", LOCATION_POLYGON_SETTINGS[location_type][:name_key]).downcase
+      @location_name = region_response.dig("attributes", LOCATION_POLYGON_SETTINGS[api_location_type][:name_key]).downcase
 
-      next unless location_categories_include?(location_name)
+      next unless location_in_scope?
 
       geometry_rings = region_response.dig("geometry", "rings")
 
+      # When the ONS API returns multiple rings for a location, some of these are small islands or similar. Some of
+      # these islands are inhabited and have a school on, and others do not. (E.g. Mersea Island in Essex, Hayling Island
+      # in Hampshire, Sheppey Island in Kent, St Mary's in the Isles of Scilly.)
+      #
       # If algolia searches by polygon are slow, (some of) these boundaries could be downsampled significantly.
-
-      ring_index = if location_type == :counties
-                     DOWNCASE_COUNTIES_WITH_RING_INDICES[location_name]
-                   else
-                     0
-                   end
-
-      points = []
-      geometry_rings[ring_index].each do |point|
-        # API returns coords in an unconventional lng,lat order
-        # Coordinates rounded as they are stored as double precision floats which have a precision of
-        # 15 decimal digits. All UK coordinates only have a maximum of 2 digits before decimal point.
-        points.push(*point.reverse.map { |coord| coord.round(13) })
+      #
+      # Store them in a hash because Postgres doesn't support multidimensional arrays of varying extents.
+
+      polygons = []
```

Comment:
Can we do something this?

```ruby
polygons = geometry_rings.map { |ring| ring.map { |point| *point.reverse.map { |coord| coord.round(13) } } }
```

Although now I've written it the map-ception is confusing me and the each blocks are probably clearer 😂